### PR TITLE
ARXIVNG-2022 add scopes required for submission to the default auth set

### PR DESCRIPTION
**Pull request by erickpeirson**
This adds 

```
    # Allows usage of the compilation service during submission.
    READ_COMPILE,
    CREATE_COMPILE,

    # Allows usage of the file management service during submission.
    READ_UPLOAD,
    WRITE_UPLOAD,
    DELETE_UPLOAD_FILE,
    
    # "Soft" deletion -- makes submission/request inactive.
    DELETE_SUBMISSION
```

to the default scopes granted an authenticated user.

Are we missing anything?

### CODE REVIEW

File: users/arxiv/users/auth/scopes.py [hunk 1]
```diff
@@ -77,24 +77,55 @@
...
+"""Can delete files from a file management upload workspace."""
 
 READ_UPLOAD_LOGS = Scope(Scope.domains.UPLOAD, 'read_logs')
+"""Can read logs for a file management upload workspace."""
+
 READ_UPLOAD_SERVICE_LOGS = Scope(Scope.domains.UPLOAD, 'read_service_logs')
+"""Can read service logs in the file management service."""
 
 READ_COMPILE = Scope(Scope.domains.COMPILE, Scope.actions.READ)
+"""Can read products generated by the compilation service."""
```
Comment (mhl10) [inline users/arxiv/users/auth/scopes.py]: Minor. Do we expect the compilation service to compile anything besides documents?
```suggestion
"""Can read documents generated by the compilation service."""
```

Comment (erickpeirson) [reply users/arxiv/users/auth/scopes.py]: Good catch; nope!

File: users/arxiv/users/auth/scopes.py [hunk 2]
```diff
@@ -77,24 +77,55 @@
...
+"""Can read logs for a file management upload workspace."""
+
 READ_UPLOAD_SERVICE_LOGS = Scope(Scope.domains.UPLOAD, 'read_service_logs')
+"""Can read service logs in the file management service."""
 
 READ_COMPILE = Scope(Scope.domains.COMPILE, Scope.actions.READ)
+"""Can read documents generated by the compilation service."""
+
 CREATE_COMPILE = Scope(Scope.domains.COMPILE, Scope.actions.CREATE)
+"""Can create new products via the compilation service."""
```
Comment (mhl10) [inline users/arxiv/users/auth/scopes.py]: Similarly:
```suggestion
"""Can create new documents via the compilation service."""
```

File: users/arxiv/users/auth/scopes.py [hunk 3]
```diff
@@ -77,24 +77,55 @@
...
+"""Can read service logs in the file management service."""
 
 READ_COMPILE = Scope(Scope.domains.COMPILE, Scope.actions.READ)
+"""Can read documents generated by the compilation service."""
+
 CREATE_COMPILE = Scope(Scope.domains.COMPILE, Scope.actions.CREATE)
+"""Can create new documents via the compilation service."""
 
 READ_FULLTEXT = Scope(Scope.domains.FULLTEXT, Scope.actions.READ)
+"""Can access plain text content extracted from e-prints."""
```
Comment (mhl10) [inline users/arxiv/users/auth/scopes.py]: Another minor one:
```suggestion
"""Can access plain text extracted from compiled documents."""
```

File: users/arxiv/users/auth/scopes.py [hunk 4]
```diff
@@ -77,24 +77,55 @@
...
+
 CREATE_FULLTEXT = Scope(Scope.domains.FULLTEXT, Scope.actions.CREATE)
+"""Can trigger new plain text extractions from e-prints."""
```
Comment (mhl10) [inline users/arxiv/users/auth/scopes.py]:
```suggestion
"""Can trigger new plain text extractions from compiled documents."""
```

File: users/arxiv/users/auth/scopes.py [hunk 5]
```diff
@@ -77,24 +77,55 @@
...
+
+    # Allows usage of the file management service during submission.
+    READ_UPLOAD,
+    WRITE_UPLOAD,
+    DELETE_UPLOAD_FILE
 ]
+"""
+The default scopes afforded to an authenticated user.
+
+This static list be deprecated by RBAC at a later milestone of arXiv-NG.
```
Comment (mhl10) [inline users/arxiv/users/auth/scopes.py]:
```suggestion
This static list will be deprecated by role-based access control (RBAC) at a later milestone of arXiv-NG.
```

**Comment by mhl10** [comment]
How about `REMOVE_SUBMISSION`?

**Comment by erickpeirson** [comment]
> How about `REMOVE_SUBMISSION`?

Yeah, we definitely need that; we appear to be using ``DELETE_SUBMISSION`` here:
https://github.com/arXiv/arxiv-submission-ui/blob/develop/submit/routes/ui.py#L179

**Comment by erickpeirson** [comment]
> > Yeah, we definitely need that; we appear to be using `DELETE_SUBMISSION` here:
> > https://github.com/arXiv/arxiv-submission-ui/blob/develop/submit/routes/ui.py#L179
> 
> A follow-up not germane to this PR: do we need to semantically differentiate between states of deletion/removal? Heretofore our notion of deletion with respect to submissions has been soft (i.e. it's never truly deleted except for source files).

At this point we haven't gone any further than the classic system in that respect. Events are immutable, so in a way we've doubled down on "soft" deletion. But indeed, that is a slightly different concept than deletion in the file manager context. Maybe that's OK?

**Comment by DavidLFielding** [comment]
Looks good.

**Pull request merged**